support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go

```diff
@@ -79,28 +81,32 @@ type IdentityMaps struct {
 // PodReconciler holds the runtime configuration and state of this controller
 type PodReconciler struct {
 	client.Client
-	c             PodReconcilerConfig
-	identityTempl *template.Template
+	c                PodReconcilerConfig
+	identityTempl    *template.Template
+	dnsNameTemplates []*template.Template
 }
 
 // NewPodReconciler creates a new PodReconciler object
 func NewPodReconciler(config PodReconcilerConfig) (*PodReconciler, error) {
 	if config.IdentityTemplate == "" && config.PodAnnotation == "" && config.PodLabel == "" {
 		config.IdentityTemplate = DefaultSpiffeIDPath
 	}
-	if config.IdentityTemplate == "" {
-		return &PodReconciler{
-			Client: config.Client,
-			c:      config,
-		}, nil
-	}
 
-	tmpl, err := template.New("IdentityTemplate").Parse(config.IdentityTemplate)
+	identityTempl, dnsNameTemplates, err := validateTemplates(config)
 	if err != nil {
-		config.Log.WithError(err).WithField("identity_template", config.IdentityTemplate).Error("error parsing identity template")
-		return &PodReconciler{}, fmt.Errorf("parsing identity template: %w", err)
+		return &PodReconciler{}, err
 	}
-	// While the Context is persitent and can be tested here, PodInfo is dynamic and changes with each Pod update, so using a dummy entry.
+
+	return &PodReconciler{
+		Client:           config.Client,
+		c:                config,
+		identityTempl:    identityTempl,
+		dnsNameTemplates: dnsNameTemplates,
+	}, nil
+}
+
+func validateTemplates(config PodReconcilerConfig) (*template.Template, []*template.Template, error) {
```

Comment:
nit: This function also parses the template, so maybe a better name would be `validateAndParseTemplates()`? Or split the validate and parse parts into separate functions.

Reply:
I agree .. splitting this up would be great, I didn't comment on it but the function is a little large and hard to grok